### PR TITLE
Spacing Sizes Control: Try improving layout spacing

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
## What?

Fixes a few spacing and size issues within the spacing sizes control. It also upsizes the inputs to 40px while we have the chance (also made it easier to address some of the spacing tweaks).

## Why?

We all want a clean, tidy and aligned UI.

## How?

- Prevents the grid template row being too tall for the intended 16px labels
- Remove 16px height styles squashing icon buttons
- Apply a negative margin to icon buttons to not enlarge the intended 16px label row
- Removed some unnecessary and duplicate styles
- Make the range controls vertically center within the 40px height wrapper previously given
- Upsize the custom input UnitControl and RangeControl so they are 40px as well
- Update margins so there's an 8px gap between the label row and inputs & 16px gap between inputs and next side label

## Testing Instructions

1. Test the control in the site editor; e.g. Global Styles > Blocks > Group > Layout
2. In the post editor, try it out for a block that has spacing support e.g. Group
3. Inspecting via dev tools you can get a better view of what spacing has been applied. Crosscheck with Figma
4. Update your theme.json such that your spacing presets has `> 8` steps. See `settings.spacing.spacingScale.steps`.
5. Confirm the custom select control has consistent spacing and height
6. Confirm visual appearance across the usual culprits: Chrome, Safari, Firefox etc.

__Note: There as some horizontal scrolling issues with the sidebar in Firefox and Safari but they occur on trunk and look unrelated to these changes__

## Screenshots or screencast 

https://user-images.githubusercontent.com/60436221/195009843-e20681b4-5939-4409-9280-ddb2020fe27c.mp4

### CODE REVIEW

**Comment by jasmussen** [comment]
Looks good, no longer looks skewed, and there's a gutter between slider and input!

Screenshot 2022-10-11 at 08 53 53

Very probably separate to this particular PR, so not a blocker, but it would be good to have a 16px gutter between input and slider. It's okay that the knob "overlaps" the gutter, as it sits on the far left of the slider. But in a knobless slider the gap between [should be 16px](https://github.com/WordPress/gutenberg/issues/43423#issuecomment-1232540038). As noted, probably separate and part of 43423.

Thank you!

**Review by jasmussen — APPROVED** [review]
Would love a check by @jameskoster as well, but giving a preliminary apprpoval.

**Comment by aaronrobertshaw** [comment]
Thanks for the quick review @jasmussen, the paint was barely dry on it 🚀 

> but it would be good to have a 16px gutter between input and slider

The styles for this component set the margin providing the gap and I already tweaked them once to be more consistent with other controls like this. I've just pushed a tweak to update these to have the 16px gap. We can address any other UnitControl/RangeControl combos as follow-ups.

> Would love a check by @jameskoster as well, but giving a preliminary approval.

I'll keep an eye out for any feedback @jameskoster has and will iron that out tomorrow.

**Comment by jasmussen** [comment]
I think you can land this when the tests pass, if it has to go quick, honestly. But if it isn't urgent, there's time for brewing the morning coffee! 😅

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (ciampo) [inline packages/block-editor/src/components/spacing-sizes-control/style.scss:96]: Any reasons why we're using the internal classname (`components-range-control`) instead of the classname assigned specifically by `SpacingSizesControl` (i.e. `components-spacing-sizes-control__custom-value-range`) ?

**Comment by ciampo** [comment]
> We can address any other UnitControl/RangeControl combos as follow-ups.

Sounds good to me — I'd love to have a detailed assessment of any discrepancies / gaps between the design specs and the current implementations for these components

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (aaronrobertshaw) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:96]: Mainly just that it was the only common classname across all range controls within the `SpacingSizesControl`.  The classname you mentioned is only used for the custom spacing values. There's also the preset slider.

I can replace the `components-range-control` with multiple selectors.

Comment (aaronrobertshaw) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:96]: While taking a look at switching this out, I noticed there are already other uses of internal RangeControl class names to be able to tweak the marks etc into the design for this control. I'll hold off on replacing this for now.

**Comment by aaronrobertshaw** [comment]
I've just pushed an additional tweak to ensure the custom select control also has a consistent margin and height. Setting the `spacingScale.steps` in theme.json to `10` is an easy way to trigger this view in the component to test.

Screen Shot 2022-10-11 at 8 16 21 pm

**Review by jameskoster — APPROVED** [review]
Much better :)

Before:

https://user-images.githubusercontent.com/846565/195085012-1e767cec-03cf-45f8-855a-9f573b139f92.mp4



After:

https://user-images.githubusercontent.com/846565/195085039-13ee1246-1385-44fe-a2df-9c18b399e994.mp4

File: packages/block-editor/src/components/spacing-sizes-control/style.scss
Comment (ciampo) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:96]: Makes sense! I guess at most we can add a note (in code or anywhere we track the follow-up work) regarding the refactor of the `components-range-control` classname.

Comment (aaronrobertshaw) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:96]: The same sort of tidy-up will need to happen for the `BorderRadiusControl` as well. 

I took the liberty of adding this to https://github.com/WordPress/gutenberg/issues/40507 where the replacement of the RangeControl with new slider components would happen. Makes sense to me that we refactor or clean this up only once. Of course, if that is too far out we could move that to a different issue.

Comment (ciampo) [reply packages/block-editor/src/components/spacing-sizes-control/style.scss:96]: Sounds good to me, thank you!

**Comment by glendaviesnz** [comment]
Thanks for looking at this @aaronrobertshaw. Ideally, we need to backport this to 6.1 as currently, this is a UI bug in the 6.1 RC1 release ... but is it ok for the 40px height unit controls to go in the 6.1 release as they will be different to the ones in the likes of the border control?

**Comment by aaronrobertshaw** [comment]
> but is it ok for the 40px height unit controls to go in the 6.1 release as they will be different to the ones in the likes of the border control?

That's a good question.

My understanding, at least for Gutenberg, was that it was ok to [update these panel by panel](https://github.com/WordPress/gutenberg/issues/40893#issuecomment-1242751383). Whether that still applies for the WordPress 6.1 release, I'm not sure. 

While the border controls would still be the shorter height, until https://github.com/WordPress/gutenberg/pull/41860 lands and makes its way into core, the typography and color controls are already 40px tall.

@jasmussen or @jameskoster would you be able to confirm that introducing the 40px high spacing controls is ok for 6.1?

**Review by andrewserong — COMMENTED** [review]
Just took this for a quick spin, and it's looking really nice!

Probably not a blocker, since this also appears to be the case on `trunk`, but if I set my browser to display scrollbars (to try to mimic how things might work on Windows), in the 100% position, the tooltip on the thumb appears to introduce a horizontal scrollbar:

image

I was wondering if there needs to be a wider gutter on the right hand side of the slider to account for this, or would that mess up how the bar looks in relation to right aligned items near it?

**Comment by jasmussen** [comment]
We'll want 40px almost everywhere, and it's already present in some cases not others. It'd be nice to get them all to 40px soon (give or take some edgecases we're thinking of), but of course that's a journey. Since we're already mixing, seems okay to continue.

It would be nice to understand the horizontal scrollbar, though, I missed that, sorry. Though I would consider that bugfix territory, and seems trivial.

**Comment by aaronrobertshaw** [comment]
> Probably not a blocker, since this also appears to be the case on trunk, but if I set my browser to display scrollbars (to try to mimic how things might work on Windows), in the 100% position, the tooltip on the thumb appears to introduce a horizontal scrollbar:

I believe this issue has come up before. It is due to the way the RangeControl renders its tooltips. They are a custom component, as in they don't render via `Tooltip` and its popovers. This was likely to make positioning them easier.

The custom tooltips get absolutely positioned and I'm not sure we can rely on a known width for the tooltip contents. The result being, even if we added a gutter to the right the tooltip width could still just be long enough to extend beyond the bounds of the component forcing scrollbars.

One of the items on the [list for the new SliderControl ](https://github.com/WordPress/gutenberg/issues/40507#issue-1210807390)component is to replace these custom tooltips with a regular `Tooltip` so we can enforce they stay within the component's container or the viewport etc. The work there is ongoing but a little lower down the current list of priorities.

As mentioned above, this doesn't have to be a blocker for these improvements.

> Though I would consider that bugfix territory, and seems trivial

It "seems" that way 😄

**Review by andrewserong — APPROVED** [review]
That all makes sense to me, thanks for following-up, @aaronrobertshaw. And once this PR lands, we can always try out updates that aren't targeted to 6.1, if need be, too 👍

Just took this for another spin, and it's all testing well and looking good to me:

✅ Spacing / styling looks good in post and site editors with `steps` set to 7 or lower
✅ Spacing / styling looks good in post and site editors with `steps` set to > 7, with spacing looking consistent in the select dropdowns, and the custom font size views.

LGTM!

**Review by glendaviesnz — APPROVED** [review]
The sizing works correctly now for me between preset and custom sizing

**Comment by ciampo** [comment]
Is it too late to backport this for 6.1 ? (edit: just seen the "backport" label, nevermind)

**Comment by michalczaplinski** [comment]
I just cherry-picked this PR to the wp/6.1-rc-2 branch to get it included in the next release: 346ff74d91

**Pull request merged**
